src/Writer/CSV/Options.php

```diff
@@ -9,6 +9,8 @@ final class Options
     public string $FIELD_DELIMITER = ',';
     public string $FIELD_ENCLOSURE = '"';
     public bool $SHOULD_ADD_BOM = true;
+    public string $DATE_TIME_FORMAT = DATE_ATOM;
+    public string $DATE_INTERVAL_FORMAT = '%y years, %m months, %d days, %h hours, %i minutes, %s seconds, %f microseconds';
```

Comment:
Thank you very much for investing your time in this library.

Regarding the formatting, I would like to avoid this time of customization because it doesn't preserve the possibility to easily read the CSV back with the reader.

So my proposal is to:

1. **not** allow any custom format for these fields; if the user want a custom format, simply iterate over the row cells and manually convert `DateTimeCell|DateIntervallCell` into a `StringCell`
2. default and only format for `DateTimeCell` is `DATE_ATOM`
3. default and only format for `DateIntervalCell` is `ISO-8601`, which doesn't come natively with PHP but [it can be easily coded](https://stackoverflow.com/questions/33787039/format-dateinterval-as-iso8601), see https://3v4l.org/VFN7N

Reply:
And how does it work excactly? I thought that if you want to read it back it will be string anyway because there is no way how to recognized that some text in CSV is date or date interval except of testing if text can be parsed via these formats and this is not happening right now isnt it?

I can change it if you are sure it is correct way, but from my perspective it make sense to have option to customize writer like this.

Reply:
> I thought that if you want to read it back it will be string anyway because there is no way how to recognized that some text in CSV is date or date interval except of testing if text can be parsed via these formats and this is not happening right now

You are indeed correct, but with `DATE_ATOM` and `ISO-8601` formats the user can easily translate the texts back to the corresponding PHP object

> I can change it if you are sure it is correct way, but from my perspective it make sense to have option to customize writer like this.

Please do it without customization options: we will add them later if really needed

Reply:
I remove options however, there is error from static analysis and I dont know why. It rejects that new types. It does not make sense to me. Because it prints that another type is expected.

```
::error file=src/Writer/CSV/Writer.php,line=,col=0::Ignored error pattern #^Parameter \#1 \$callback of function array_map expects \(callable\(OpenSpout\\Common\\Entity\\Cell\)\: mixed\)\|null, Closure\(OpenSpout\\Common\\Entity\\Cell\\BooleanCell\|OpenSpout\\Common\\Entity\\Cell\\EmptyCell\|OpenSpout\\Common\\Entity\\Cell\\FormulaCell\|OpenSpout\\Common\\Entity\\Cell\\NumericCell\|OpenSpout\\Common\\Entity\\Cell\\StringCell\)\: string given\.$# in path /var/www/html/src/Writer/CSV/Writer.php was not matched in reported errors.
::error file=src/Writer/CSV/Writer.php,line=52,col=0::Parameter #1 $callback of function array_map expects (callable(OpenSpout\Common\Entity\Cell): mixed)|null, Closure(OpenSpout\Common\Entity\Cell\BooleanCell|OpenSpout\Common\Entity\Cell\DateIntervalCell|OpenSpout\Common\Entity\Cell\DateTimeCell|OpenSpout\Common\Entity\Cell\EmptyCell|OpenSpout\Common\Entity\Cell\FormulaCell|OpenSpout\Common\Entity\Cell\NumericCell|OpenSpout\Common\Entity\Cell\StringCell): string given.
```

Do you have any idea how to deal with it?